### DISCUSSION TITLE
UI customisation and login without UI

### DISCUSSION THREAD

**Discussion by Makkrich** [fief-dev/fief#124] [Q&A] [answered]
Hi everyone, I have 2 question:
- Is it possible to customize the appearance of the login and admin pages?
- Is it possible to log in without redirects, using API? That is, if you throw data about the user and credentials of the client on some endpoint and gets token instead? 
Thank you in advance

**Comment by fief-bailiff** [comment]
Hail, @Fan4ik20 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
Hi @Fan4ik20 👋 

* Currently, it's not possible to customize authentication pages, but it's the next feature we plan to work on!
* So far, we made the choice to implement an authentication process based on OAuth2, as it's the easiest way to integrate an external auth system to any project. We don't currently plan to implement an API to make direct login requests, but that may be something for the long term!

**Reply by Makkrich** [reply]
Thank you very much for your reply, have a nice day

**Reply by heshdotcc** [reply]
Hey! Thanks for such an amazing project. Suppose I already have implemented a design system which I would like to preserve for authentication routes, so there would be no way to use Fief? Thanks in advance. Cheers,

**Reply by frankie567** [reply]
@hesh-cc  Indeed, you have to use the built-in authentication pages for now. That said, theme customization is now available and you are able to match those pages with your branding: https://docs.fief.dev/configure/customization/themes/
